### PR TITLE
Minor bugfixes and improvements

### PR DESCRIPTION
**Pull request by zzeekk**
- fix dbutils-api scala version with property scala.minor.version
- improve exception message if key of DataObject or Connection is not found
- implement SQLQueryExpectation
- general improvements for Expectations
- fix NPE in ExpectationDefaultImpl
- make fail simulation on missing inputSubFeeds configurable
- make more from sdl-core public

### CODE REVIEW

File: sdl-core/src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala [hunk 1]
```diff
@@ -288,32 +288,37 @@ abstract class SmartDataLakeBuilder extends SmartDataLakeLogger {
...
     (subFeeds.map(_.asInstanceOf[SparkSubFeed]), stats)
   }
 
   /**
-   * Starts a simulation run and registers all SDL first class objects that are defined in the config file which path is defined in parameter appConfig
+   * Similar to [[startSimulation]], but Actions and DataObjects are parsed from config files defined through appConfig.configuration attribute.
    */
-  def startSimulationWithConfigFile(appConfig: SmartDataLakeBuilderConfig, initialSubFeeds: Seq[SparkSubFeed], dataObjectsState: Seq[DataObjectState] = Seq())(session: SparkSession): (Seq[SparkSubFeed], Map[RuntimeEventState,Int]) = {
+  def startSimulationWithConfigFile(appConfig: SmartDataLakeBuilderConfig, initialSubFeeds: Seq[SparkSubFeed], dataObjectsState: Seq[DataObjectState] = Seq(), failOnMissingInputSubFeeds: Boolean = true)(session: SparkSession): (Seq[DataFrameSubFeed], Map[RuntimeEventState,Int]) = {
+    Environment._failSimulationOnMissingInputSubFeeds = Some(failOnMissingInputSubFeeds)
```
Comment (ddeuber) [inline sdl-core/src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala]: Why do you need to set `Environment._failSimulationOnMissingInputSubFeeds` here when it is set again in `startSimulation`?

File: sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/Expectation.scala [hunk 2]
```diff
@@ -133,6 +133,7 @@ abstract class ExpectationDefaultImpl extends Expectation {
     import org.apache.spark.sql.functions._
     expectation.map { expectationStr =>
       val valueStr = value match {
+        case null => "null"
```
Comment (ddeuber) [inline sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/Expectation.scala]: Are there other possibilities than passing null around in the metrics? I get paranoid when I see null values in Scala :)

**Review by ddeuber — APPROVED** [review]

File: sdl-core/src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala (hunk 1 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/app/SmartDataLakeBuilder.scala]: Correct, i removed it.

File: sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/Expectation.scala (hunk 2 above)
Comment (zzeekk) [reply sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/Expectation.scala]: You're right, not nice in Scala.
Background: the null value is from a Spark df.collect.
I fixed it and convert null to None.

**Pull request merged**